pkg/performanceprofile/controller/performanceprofile/components/tuned/tuned_test.go

```diff
@@ -51,87 +50,148 @@ var _ = Describe("Tuned", func() {
-			By("Populating empty additional kernel arguments cmdline")
-			Expect(manifest).To(ContainSubstring("cmdline_additionalArg="))
-			By("Populating realtime cmdline")
-			Expect(cmdlineWithRealtime.MatchString(manifest)).To(BeTrue())
+			Expect(isolated.Key("isolated_cores").String()).Should(Equal("4-5"))
+
+			bootLoader, err := tunedData.GetSection("bootloader")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(bootLoader.Key("cmdline_cpu_part").String()).Should(Equal(cmdlineCPUsPartitioning))
+			Expect(bootLoader.Key("cmdline_hugepages").String()).Should(Equal(cmdlineHugepages))
+		})
+
+		Context("default profile default tuned", func() {
+			It("should [cpu] section in tuned", func() {
+				tunedData := getTunedStructuredData(profile)
+				cpuSection, err := tunedData.GetSection("cpu")
+				Expect(err).ToNot(HaveOccurred())
+				Expect((cpuSection.Key("force_latency").String())).Should(Equal("cstate.id:1|3"))
+				Expect((cpuSection.Key("governor").String())).Should(Equal("performance"))
+				Expect((cpuSection.Key("energy_perf_bias").String())).Should(Equal("performance"))
+				Expect((cpuSection.Key("min_perf_pct").String())).Should(Equal("100"))
+			})
 		})
 
 		When("realtime hint disabled", func() {
-			BeforeEach(func() {
+			It("should not contain realtime related parameters", func() {
 				profile.Spec.WorkloadHints = &performancev2.WorkloadHints{RealTime: pointer.BoolPtr(false)}
+				tunedData := getTunedStructuredData(profile)
+				bootLoader, err := tunedData.GetSection("bootloader")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(bootLoader.Key("cmdline_realtime").String()).ShouldNot(Equal(cmdlineRealtime))
 			})
+		})
 
-			It("should not contain realtime related parameters", func() {
-				manifest := getTunedManifest(profile)
-				Expect(manifest).ToNot(ContainSubstring("service.stalld=start,enable"))
-				Expect(manifest).ToNot(ContainSubstring("sched_rt_runtime_us=-1"))
-				Expect(manifest).ToNot(ContainSubstring("kernel.hung_task_timeout_secs=600"))
-				Expect(manifest).ToNot(ContainSubstring("kernel.sched_rt_runtime_us=-1"))
-				By("Populating realtime cmdline")
-				Expect(cmdlineWithRealtime.MatchString(manifest)).ToNot(BeTrue())
+		When("realtime hint enabled", func() {
+			It("should contain realtime related parameters", func() {
+				profile.Spec.WorkloadHints = &performancev2.WorkloadHints{RealTime: pointer.BoolPtr(true)}
+				tunedData := getTunedStructuredData(profile)
+				service, err := tunedData.GetSection("service")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(service.Key("service.stalld").String()).Should(Equal("start,enable"))
+				scheduler, err := tunedData.GetSection("scheduler")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(scheduler.Key("sched_rt_runtime_us").String()).Should(Equal("-1"))
+				sysctl, err := tunedData.GetSection("sysctl")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(sysctl.Key("kernel.hung_task_timeout_secs").String()).Should(Equal("600"))
```

Comment:
According to: https://github.com/onsi/gomega/blob/master/gomega_dsl.go#L415  ```To```/```ToNot```/```NotTo``` are used with ```Expect```